cpp/src/connection_options.cpp

```diff
@@ -175,7 +175,8 @@ class connection_options::impl {
             }
         } else if (!client && ssl_server_options.set) {
             pn_ssl_t *ssl = pn_ssl(pnt);
-            if (pn_ssl_init(ssl, ssl_server_options.value.impl_->pn_domain(), NULL)) {
+            pn_ssl_domain_t* ssl_domain = ssl_server_options.value.impl_ ? ssl_server_options.value.impl_->pn_domain() : NULL;
+            if (pn_ssl_init(ssl, ssl_domain, NULL)) {
```

Comment:
This change is wrong unfortunately.
the comparison with the client case is misleading. `pn_ssl_init(ssl, NULL, NULL)` will create an ssl connection with the default domain - however the default domain is a *client* domain so you can't do this if the connection is a server.
So you need to replace he NULL with something like `pn_ssl_domain(PN_SSL_MODE_SERVER)`.

Reply:
It is a shame that this wasn't picked up by any tests - I thought we had some ssl server tests.

Reply:
@astitcher  I replaced NULL by pn_ssl_domain(PN_SSL_MODE_SERVER) like you mentioned